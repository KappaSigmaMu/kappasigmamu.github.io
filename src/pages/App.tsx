import React, { useState } from 'react'
import { Container } from 'react-bootstrap'
import { BrowserRouter, Routes, Route } from 'react-router-dom'
import styled, { ThemeProvider } from 'styled-components'
import { Navbar } from '../components/Navbar'
import { GlobalStyle } from '../styles/globalStyle'
import { Theme } from '../styles/Theme'
import { SubstrateContextProvider, useSubstrate } from '../substrate'
import { CyborgGuide } from './CyborgGuide'
import { Home } from './home/Home'
import { LandingPage } from './LandingPage'
import { Welcome } from './Welcome'

const Main = () => {
  const { apiState, apiError } = useSubstrate()
  const [activeAccount, setActiveAccount] = useState<string>('')
  const [accounts, setAccounts] = useState<{ name: string | undefined; address: string }[]>([])

  const loader = (text: string) => {
    return <p>{text}</p>
  }

  if (apiState === 'CONNECTING') return loader('Connecting')
  if (apiState === 'ERROR') return loader(`${JSON.stringify(apiError, null, 4)}`)
  if (apiState !== 'READY') return loader('Connecting')

  return (
    <>
      <GlobalStyle />
      <StyledMain fluid>
        <BrowserRouter>
          <Routes>
            <Route path="/" element={
              <>
                <Navbar
                  accounts={accounts}
                  activeAccount={activeAccount}
                  setAccounts={setAccounts}
                  setActiveAccount={setActiveAccount}
                  showAccount
                  showSocialIcons
                />
                <LandingPage activeAccount={activeAccount} />
              </>
            }/>
            <Route path="/cyborg-guide" element={
              <>
                <Navbar
                  accounts={accounts}
                  activeAccount={activeAccount}
                  setAccounts={setAccounts}
                  setActiveAccount={setActiveAccount}
                  showAccount
                  showSocialIcons
                  showGalleryButton
                />
                <CyborgGuide />
              </>
            }/>
            <Route path="/welcome" element={
              <>
                <Navbar
                  accounts={accounts}
                  activeAccount={activeAccount}
                  setAccounts={setAccounts}
                  setActiveAccount={setActiveAccount}
                  showAccount
                  showBrandIcon
                  showGalleryButton
                />
                <Welcome />
              </>
            }/>
            <Route path="/home" element={
              <>
                <Navbar
                  accounts={accounts}
                  activeAccount={activeAccount}
                  setAccounts={setAccounts}
                  setActiveAccount={setActiveAccount}
                  showAccount
                  showBrandIcon
                  showGalleryButton
                />
                <Home activeAccount={activeAccount} />
              </>
            }/>
          </Routes>
        </BrowserRouter>
      </StyledMain>
    </>
  )
}

const App = () => {
  return (
    <SubstrateContextProvider>
      <ThemeProvider theme={Theme}>
        <Main />
      </ThemeProvider>
    </SubstrateContextProvider>
  )
}

const StyledMain = styled(Container)`
`

export { App }
