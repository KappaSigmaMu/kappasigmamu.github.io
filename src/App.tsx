import { Container, Nav, Navbar, Button } from 'react-bootstrap'
import { Switch, Route, Link } from 'react-router-dom'
import NavLogo from '../src/static/nav-logo.png'
import { About } from './pages/About'
import { Blog } from './pages/Blog'
import { CyborgJourney } from './pages/CyborgJourney'
import { Home } from './pages/Home'

function App() {
  return (
    <Container>
      <Navbar bg="black" variant="dark">
        <Container fluid>
          <Navbar.Brand as={Link} to="/">
            <img
              src={NavLogo}
              alt="KappaSigmaMu"
              width="150"
              className="d-inline-block align-top"
            />
          </Navbar.Brand>
          <Nav className="mx-auto">
            <Nav.Link as={Link} to="/">
              Kusama Network
            </Nav.Link>
            <Nav.Link as={Link} to="/about">
              About
            </Nav.Link>
            <Nav.Link as={Link} to="/blog">
              Blog
            </Nav.Link>
            <Nav.Link as={Link} to="/cyborg-journey">
              Cyborg Journey
            </Nav.Link>
          </Nav>
          <Button>Connect Wallet</Button>
        </Container>
      </Navbar>

      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route path="/about">
          <About />
        </Route>
        <Route path="/blog">
          <Blog />
        </Route>
        <Route path="/cyborg-journey">
          <CyborgJourney />
        </Route>
      </Switch>
    </Container>
  )
}

export { App }
