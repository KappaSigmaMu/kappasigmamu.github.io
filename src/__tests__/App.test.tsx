import { render, screen } from '@testing-library/react'
import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { App } from '../App'

test('renders App', () => {
  render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  )

  const element = screen.getByText(/KappaSigmaMu Society/i)
  expect(element).toBeInTheDocument()
})
