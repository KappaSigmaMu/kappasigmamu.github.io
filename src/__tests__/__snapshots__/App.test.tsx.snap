// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<Router
  action="POP"
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <App />
</Router>
`;
